kernelci/rootfs.py

```diff
@@ -70,12 +88,34 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._frag = 'baseline'  # ToDo: add to configuration
 
-    def build(self, config, data_path, arch):
-        cmd = 'cd {data_path} && ./configs/frags/build {arch} {frag}'.format(
-            data_path=data_path,
-            arch=arch,
-            frag=self._frag
-        )
+    def build(self, config, data_path, arch, output):
+        absoutput_dir = os.path.abspath(output)
+        artifact_dir = os.path.join(absoutput_dir, '_install_',
+                                    self.name, arch)
+        temp_dir = os.path.join(absoutput_dir, 'temp')
+
+        # Create directories if missing
+        if not os.path.isdir(artifact_dir):
+            os.makedirs(artifact_dir, exist_ok=True)
+        if not os.path.isdir(temp_dir):
+            os.makedirs(temp_dir, exist_ok=True)
+        os.chdir(temp_dir)
```

Comment:
This works, but relying on the current directory adds some implicit state to the logic so it's more likely to break with future changes. Ideally, each command should be run within a full relative path from the top-level directory.  See the `kernelci.build` code, there's `cd {kdir}` at the beginning of each command. Maybe this could be done here as a follow-up PR.